### ISSUE TITLE
Passing replicas as overrides in kustomize

### ISSUE THREAD

**Issue by meenaravichandran1** [argoproj/argo-cd#15858]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

I am trying to pass replicas to a kustomize app in the below format and it throws the error in the Argocd server.

**Payload**:
```
{
    "apiVersion": "argoproj.io/v1alpha1",
    "kind": "Application",
    "spec": {
        "project": "09fd9492",
        "source": {
            "repoURL": "https://github.com/meenaravichandran/meenasync.git",
            "path": "example/base",
            "targetRevision": "kustomize",
            "kustomize": {
                "namePrefix": "prod-",
                "nameSuffix": "test",
                "replicas": [
                    {
                        "name": "prod-myapp-deploymenttest",
                        "count": "5"
                    }
                ]
            }
        },
        "destination": {
            "server": "https://kubernetes.default.svc",
            "namespace": "default"
        }
    },
    "status": {}
}
```

**Error**:
```
time="2023-10-05T18:09:26Z" level=info msg="received unary call /application.ApplicationService/Update" grpc.method=Update grpc.request.content="%!v(PANIC=String method: reflect.Value.Bytes of non-byte slice)" grpc.service=application.ApplicationService grpc.start_time="2023-10-05T18:09:26Z" span.kind=server system=grpc
W1005 18:09:28.401231       1 warnings.go:70] unknown field "spec.source.kustomize.replicas"
time="2023-10-05T18:09:28Z" level=info msg="admin updated application spec" application=kus dest-namespace=default dest-server="https://kubernetes.default.svc" reason=ResourceUpdated type=Normal user=admin
```

If I set it in the kustomization.yaml and sync the application it is working as expected. But if I send it via the API, it is not working.

I also noticed that the replicas is not showing up as overrides in the kustomize, it just affects the total number of pods.

**To Reproduce**

1. Create a kustomize app
2. Modify its replicas via the API

**Expected behavior**

The replica count should reflect in the application.

**Screenshots**

**Version**

```shell
v2.7.13
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by crenshaw-dev** [comment]
Looks like that field was in fact available in 2.7.13. https://github.com/argoproj/argo-cd/blob/v2.7.13/docs/operator-manual/application.yaml#L94-L96

Make sure you've applied the appropriate version of the Application CRD: https://github.com/argoproj/argo-cd/blob/v2.7.13/manifests/crds/application-crd.yaml

**Comment by meenaravichandran1** [comment]
thanks @crenshaw-dev. This is working. I tried to update the argo CD version by just updating the image. Now I know that I have to update the CRDs as well. Should I take care of anything else while updating the version?
